### PR TITLE
feat: add remind functionality

### PR DESCRIPTION
**Pull request by kiram15**
# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/learner-credit-management/PendingAssignmentRemindButton.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+};
+
+PendingAssignmentRemindButton.propTypes = {
+  row: PropTypes.shape({
+    original: PropTypes.shape({
+      assignmentConfiguration: PropTypes.string.isRequired,
+      learnerEmail: PropTypes.string,
+      recentAction: PropTypes.shape({
+        actionType: PropTypes.string.isRequired,
+        timestamp: PropTypes.string.isRequired,
```
Comment (katrinan029) [inline src/components/learner-credit-management/PendingAssignmentRemindButton.jsx:47]: [question]: Do we need `recentAction` property here?

File: src/components/learner-credit-management/data/hooks/useSuccessfulReminderToastContextValue.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const [isToastOpen, setIsToastOpen] = useState(false);
+  const [assignmentUuidCount, setAssignmentUuidCount] = useState();
+
+  const handleDisplayToast = useCallback((assignmentUuids) => {
+    setIsToastOpen(true);
+    setAssignmentUuidCount(assignmentUuids);
+  }, []);
+
+  const handleCloseToast = useCallback(() => {
+    setIsToastOpen(false);
```
Comment (katrinan029) [inline src/components/learner-credit-management/data/hooks/useSuccessfulReminderToastContextValue.js:20]: [suggestion]: I think we should a unit for this. There's an example in `BudgetDetailPageWrapper.tests.jsx` that tests  something similar for the cancel toast.

File: src/components/learner-credit-management/data/hooks/useRemindContentAssignments.test.jsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    expect(result.current).toEqual({
+      assignButtonState: 'default',
+      remindContentAssignments: expect.any(Function),
+      close: expect.any(Function),
+      isOpen: false,
+      open: expect.any(Function),
+    });
+
+    await waitFor(() => act(() => result.current.remindContentAssignments()));
```
Comment (katrinan029) [inline src/components/learner-credit-management/data/hooks/useRemindContentAssignments.test.jsx:59]: [suggestion]: I think you can remove the `act` here.
`await waitFor(() => result.current.remindContentAssignments());`

File: src/components/learner-credit-management/data/hooks/useRemindContentAssignments.test.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
```
Comment (katrinan029) [inline src/components/learner-credit-management/data/hooks/useRemindContentAssignments.test.jsx:92]: [suggestion]: same comment as above

File: src/components/learner-credit-management/data/hooks/useRemindContentAssignments.test.jsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
```
Comment (katrinan029) [inline src/components/learner-credit-management/data/hooks/useRemindContentAssignments.test.jsx:126]: [suggestion]: same here

**Review by katrinan029 — APPROVED** [review]
Looks good! I just have a few nits & suggestions.

File: src/components/learner-credit-management/RemindAssignmentModal.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ActionRow, ModalDialog, StatefulButton,
+} from '@edx/paragon';
+import { Mail } from '@edx/paragon/icons';
+import { BudgetDetailPageContext } from './BudgetDetailPageWrapper';
+
+const RemindAssignmentModal = ({
+  assignButtonState, remindContentAssignments, close, isOpen, uuidCount,
```
Comment (adamstankiewicz) [inline src/components/learner-credit-management/RemindAssignmentModal.jsx:10]: nit: `assignButtonState` -> `remindButtonState`? 🤔

File: src/components/learner-credit-management/assignments-status-chips/FailedBadEmail.jsx [hunk 7]
```diff
@@ -40,7 +40,7 @@ const FailedBadEmail = ({ learnerEmail }) => {
               </li>
               <li>
                 Get more troubleshooting help at{' '}
-                <Hyperlink destination="https://edx.org" target="_blank">
+                <Hyperlink destination="https://edx.org" showLaunchIcon target="_blank">
```
Comment (adamstankiewicz) [inline src/components/learner-credit-management/assignments-status-chips/FailedBadEmail.jsx:43]: nit: `showLaunchIcon` is the default, and could be removed (here and elsewhere):

image

File: src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx [hunk 8]
```diff
@@ -1184,7 +1203,7 @@ describe('<BudgetDetailPage />', () => {
     const cancelRowAction = screen.getByTestId('cancel-assignment-test-uuid');
     expect(cancelRowAction).toBeInTheDocument();
     if (shouldDisplayRemindAction) {
-      const remindRowAction = screen.getByTestId('remind-assignment-test-uuid');
+      const remindRowAction = screen.getByTestId('remind-learner-test-uuid');
```
Comment (adamstankiewicz) [inline src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx:1206]: [nit, note] fwiw, `remind-assignment-test-uuid` may still be more appropriate as the same learner could have multiple assignments to several courses; `assignment` implies a `learner+content` pair.

**Review by adamstankiewicz — APPROVED** [review]
Looks great! Just minor nits.

**Pull request merged**